src/regdesk/states/loading_state.js

```diff
@@ -0,0 +1,51 @@
+import { RegState } from './reg_state.js';
+
+/**
+ * Loading state
+ */
+export class LoadingState extends RegState {
+  /**
+   * Get the transition events this state can process.
+   */
+  static get events() {
+    return {
+      CANCEL: 'CANCEL',
+      SINGLE_RESULT_READY: 'SINGLE_RESULT_READY',
+      MULTIPLE_RESULTS_READY: 'MULTIPLE_RESULTS_READY',
+      NO_RESULT_READY: 'NO_RESULT_READY',
+    };
+  }
+
+  /**
+   * Initializes a new instance of the LoadingState class.
+   * @param {RegMachineArgs} regMachineArgs - Standard argument set for states.
+   * @param {*} eventMap - Mapping of this object's transition events to new states.
+   */
+  constructor(regMachineArgs, eventMap) {
+    super(regMachineArgs, eventMap);
+  }
+
+  /**
+   * Actions to perform when entering the state.
+   * @param {CustomEvent} e - The event details object.
+   */
+  enterState(e) {
+    this.leftButton.show().setTransitionCallback(this, LoadingState.events.CANCEL);
+    this.rightButton.hide();
+
+    // Demo code!
+    this.centerField.innerText = `Searching for ${e.detail.searchText}`;
```

Comment:
Ideally you shouldn't set text via javascript. Makes i18n a nightmare. You would just show/hide elements from javascript and keep all the translations in the html.

Reply:
I agree, hence the above note that this is demo code. Mercifully isn't a feature (for _this_ con anyway) but is something I attempt to keep in mind either way.

Reply:
See below, moved to a separate span element within a static html string.